src/subdomains/supporting/fiat-output/fiat-output-job.service.ts

```diff
@@ -0,0 +1,261 @@
+      return map;
+    }, new Map<string, FiatOutput[]>());
+
+    const sortedEntities = Array.from(groupedEntities.values()).map((group) =>
+      group.sort((a, b) => {
+        if (a.accountIban !== b.accountIban) return a.accountIban.localeCompare(b.accountIban);
+        if (a.type !== b.type) return a.type.localeCompare(b.type);
+        return a.amount - b.amount;
+      }),
+    );
+
+    const assets = await this.assetService
+      .getAllAssets()
+      .then((assets) => assets.filter((a) => a.type === AssetType.CUSTODY && a.bank));
+    const liqBalances = await this.liquidityManagementBalanceService.getAllLiqBalancesForAssets(
+      assets.map((a) => a.id),
+    );
+
+    for (const accountIbanGroup of sortedEntities) {
+      let updatedFiatOutputs = 0;
+
+      for (const entity of accountIbanGroup.filter((e) => !e.isReadyDate)) {
+        try {
+          const liqBalance = liqBalances.find((l) => l.asset.bank.iban === entity.accountIban);
+          const pendingBalance = accountIbanGroup.reduce(
+            (sum, tx) => sum + (tx.isReadyDate && !tx.bankTx ? tx.amount : 0),
+            0,
+          );
+          const availableBalance =
+            liqBalance.amount - pendingBalance - updatedFiatOutputs - Config.processing.fiatOutput.bankPuffer;
+
+          if (availableBalance > entity.amount) {
+            updatedFiatOutputs += entity.amount;
+            const ibanCountry = entity.iban.substring(0, 2);
+
+            if (
+              !entity.buyFiats.length ||
+              (entity.buyFiats?.[0]?.cryptoInput.isConfirmed &&
+                entity.buyFiats?.[0]?.cryptoInput.asset.blockchain &&
+                (liqBalance.asset.name !== 'CHF' || ['CH', 'LI'].includes(ibanCountry)))
+            )
+              await this.fiatOutputRepo.update(entity.id, { isReadyDate: new Date() });
+          }
+        } catch (e) {
+          this.logger.error(`Error in fiatOutput fillUp job: ${entity.id}`, e);
+        }
+      }
+    }
+  }
+
+  private async setBatchId(): Promise<void> {
+    const entities = await this.fiatOutputRepo.find({
+      where: { isReadyDate: Not(IsNull()), batchId: IsNull(), isComplete: false },
+      order: { accountIban: 'ASC', id: 'ASC' },
+    });
+
+    let currentBatch: FiatOutput[] = [];
+    for (const entity of entities) {
+      if (!currentBatch.length) currentBatch.push(entity);
+
+      const currentBatchAmount = currentBatch.reduce((sum, tx) => sum + tx.amount, 0);
+
+      if (
+        currentBatch[0].accountIban !== entity.accountIban ||
+        currentBatchAmount + entity.amount > Config.processing.fiatOutput.batchAmountLimit
+      ) {
+        await this.fiatOutputRepo.update(
+          { id: In(currentBatch.map((f) => f.id)) },
+          { batchId: (await this.getLastBatchId()) + 1, batchAmount: currentBatchAmount * 100 },
+        );
+        currentBatch = [entity];
+      } else {
+        currentBatch.push(entity);
+      }
+    }
```

Comment:
Is this supposed to only create batches if the total amount would be larger than `batchAmountLimit`?

Reply:
Maybe it would be clever to have a `batches` array which we fill while looping through the entities and save them all in the end?